pyproject.toml

```diff
@@ -57,6 +30,37 @@ configparser = "^5.3.0"
 prettytable="^3.6.0"
 
 [tool.poetry.dev-dependencies]
+mypy = "^1.3.0"
+coverage = "^7.2.7"
+flake8 = ">=5.0.4,<7.0.0"
+pytest = "^7.1.2"
+setuptools = ">=65.6.3,<68.0.0"
+
+# never used on top level, remove all?
+docker = "^6.1.1"
+tqdm = "^4.65.0"
+python-dotenv = "^1.0.0"
+attrs = "^21.4.0"
+boltons = "^21.0.0"
+bracex = "^2.3.post1"
+charset-normalizer = "^2.1.1"
+defusedxml = "^0.7.1"
+face = "^20.1.1"
+glom = "^22.1.0"
+idna = "^3.4"
+jsonschema = "^4.17.3"
+pyparsing = "^3.0.9"
+python-lsp-jsonrpc = "^1.0.0"
+six = "^1.16.0"
+typing-extensions = "^4.5.0"
+ujson = "^5.7.0"
+wcmatch = "^8.4.1"
+websocket-client = "^1.5.1"
+mypy-extensions = "^1.0.0"
+pathspec = "^0.11.1"
+platformdirs = "^3.5.0"
+tomli = "^2.0.1"
+pathos = ">=0.2.9,<0.4.0"
```

Comment:
I think these are all safe to remove. They all seem to be dependencies of dependencies rather then ones used by guarddog.

Reply:
@christophetd Any feedback on this? I do not just want to remove any of those if they are actually used. I do think this needs to be done though because I do not think the intent is to distribute pytest, docker, and six to everyone that installs guarddog.

Reply:
Makes sense, I think these are old dependencies that did not get removed. Happy to merge when CI passes

Reply:
Great. Thanks for the input. I will remove these all then. If they are still needed they will get pulled in automatically